examples/voting/src/lib.rs

```diff
@@ -244,36 +167,32 @@ fn vote(
 #[receive(
     contract = "voting",
     name = "getNumberOfVotes",
-    parameter = "VoteParameter",
-    return_value = "VoteCount"
+    parameter = "String",
+    return_value = "u32"
 )]
-fn get_votes(ctx: &ReceiveContext, host: &Host<State>) -> ReceiveResult<VoteCount> {
-    // Parse the parameter.
-    let param: VoteIndex = ctx.parameter_cursor().get()?;
+fn get_votes(ctx: &ReceiveContext, host: &Host<State>) -> ReceiveResult<u32> {
+    // Parse the vote option.
+    let vote_option: String = ctx.parameter_cursor().get()?;
 
-    // Get the number of votes from the tally.
-    let result = match host.state().tally.get(&param) {
-        Some(votes) => *votes,
-        None => 0,
-    };
+    // Count the number of votes for this option.
+    let count =
+        host.state().ballots.iter().filter(|&(_voter, option)| *option == vote_option).count();
```

Comment:
Tbf for this example, the loop will be small as there will probably be very few options.

I wasn't aware of this problem though and I think there's a great opportunity to highlight this problem through the tutorial. That could be a really good idea. It might even be done best by *intentionally* including the problem in the code and documenting it and perhaps even showcasing the problem in the tutorial. That sounds very educational to me.

Do you think that could work? After all, we already include a big disclaimer at the start of the tutorial stating that we're not encouraging its use or anything of the sort.

The alternative would be to do it in a more complicated but more proper way (but... probably we still want to include the disclaimer anyway) and then better documentation to explain why it was written like that.

Also, how many options do you think you would need before this turns problematic? Is there a way to figure that out?

Reply:
Iterations/Loops are quite an anti-pattern in smart contracts, which is why I am hesitant to include them even in examples. People start to copy-paste this code and the disclaimer gets lost potentially and the outcome can be really bad of an unbounded execution in a smart contract (e.g. contract becomes unusable).

Reply:
Pointing out that the logic has been designed to avoid unbounded loops/iterations is something we should add to the tutorial or a comment.

Reply:
Alright, I added a tally map which should keep track of the votes as votes come in, allowing for constant-time lookup of the count for each option :)